### PR TITLE
build: add required configuration for publishing to maven central

### PR DESCRIPTION
**Pull request by pk-work**
This patch ...

* ... adds meta information for license, scm or developers to the
pom.xml which is required by maven central.

* adds a task to create a test Javadoc artifact, which is required by
maven central.

* adds fixes for violations in the test Javadoc

* adds the script 'publishToMaven.sh' which can be used to add required
credentials when publishing to maven central.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ plugins {
 
 group = 'io.neonbee'
 // have to use single quotes for XMake, change as soon as new version Gradle build plugin is merged
```
Comment (kristian) [inline build.gradle:25]: shall we remove all references to "XMake" in the comments? Also maybe search for it globally

File: gradle/publishing.gradle [hunk 2]
```diff
@@ -13,6 +10,44 @@ publishing {
             artifact source: shadowJar
             artifact source: sourcesJar
             artifact source: javadocJar
+
+            pom {
+                name = 'NeonBee Core'
+                description = 'NeonBee is an open source reactive dataflow engine, a data stream processing framework using Vert.x.'
+                url = 'https://neonbee.io/'
+
+                licenses {
```
Comment (kristian) [inline gradle/publishing.gradle:19]: I saw this information is redundant in the files quite often, isn't there a way to specify this only once?

**Review by kristian — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (pk-work) [reply build.gradle]: fixed

File: gradle/publishing.gradle (hunk 2 above)
Comment (pk-work) [reply gradle/publishing.gradle:19]: I don't know how. :(

**Review by s4heid — APPROVED** [review]

**Pull request merged**